### PR TITLE
DSC extension refactored code for ARM & ASM

### PR DESCRIPTION
**Pull request by eshaparmar**
These changes mainly consist of refactored code that supports both ARM and ASM. We have moved a lot of common code for ARM/ASM at one place instead of writing it in two places.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@eshaparmar__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (esparmar). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

**Comment by stankovski** [comment]
Started on demand test run - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/35/
Started MSI build - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-msi/355/

File: src/Common/Commands.Common/Commands.Common.csproj [hunk 1]
```diff
@@ -103,12 +103,16 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\packages\Microsoft.WindowsAzure.Management.4.1.1\lib\net40\Microsoft.WindowsAzure.Management.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.WindowsAzure.Storage">
```
Comment (stankovski) [inline src/Common/Commands.Common/Commands.Common.csproj]: Looks like you've added a new DLL to the output. Please regenerate WIX file if that is the case

Comment (eshaparmar) [inline src/Common/Commands.Common/Commands.Common.csproj]: I am not sure why we need to regenerate the wix file. We haven't added any new project/assembly. Just re-using an existing one(Microsoft.WindowsAzure.Management) and added it as a reference to the Command.Common project.

**Comment by stankovski** [comment]
Started MSI build - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-msi/362/

**Pull request merged**